doc: update CI signal information in README (#8328)
containerd/containerd: README.md

Please be aware: nightly builds might have critical bugs, it's not recommended for use in production and no support provided.

> ## CI Dashboard

Reviewer: My only concern with calling this the "CI Dashboard" is that it only reflects a couple of more than 40 CI actions on the branch head or each pull request. Should we call it the "K8s CI Dashboard" since it is only the Kubernetes->CRI aspect of containerd testing?

Author: The readme already contains a badge for CI and Nightly builds.  Thats why skipped adding it here.  

But for this I think k8s-CRI also makes sense.

Author: @estesp updated the wording , can you take another look.

> ## K8s CI Dashboard

Reviewer 2: ```suggestion
## Kubernetes' (k8s) CI Dashboard Group
```

> The [k8s CI](https://testgrid.k8s.io/containerd) for containerd contains the following set of dashboards which is used to

Reviewer 2: ```suggestion
The [k8s CI dashboard group for containerd](https://testgrid.k8s.io/containerd) contains test results regarding
```

> check the health of the kubernetes CRI aspect of containerd

Reviewer 2: ```suggestion
the health of the kubernetes for a number of containerd release branches. 
```

> - [containerd-presubmits](https://testgrid.k8s.io/containerd-presubmits)

Reviewer 2: let's leave the list out since it will possibly change over time at the group destination...
